src/world/object/mod.rs

```diff
@@ -12,11 +12,15 @@ use crate::world::*;
 pub struct Object {
     pub name: Option<String>,
     pub model_name: String,
-    pub model: Arc<Model>,
+    pub model: Option<Arc<Model>>,
     pub transform: Transform,
 }
 
 impl Object {
+    pub fn set_model(&mut self, model: Arc<Model>) {
```

Comment:
I really don't like the idea of having this be an option. I think given the design that you have right now you should create an extra struct called `ObjectData` that gets deserialized and then in the second step use that to create an actual `Object`